packages/block-library/src/utils/is-block-metadata-experimental.js

```diff
@@ -2,18 +2,13 @@
  * Checks if the block is experimental based on the metadata loaded
  * from block.json.
  *
- * This function is in a separate file and uses the older JS syntax so
- * that it can be imported in both:
- * – block-library/src/index.js
- * – block-library/src/babel-plugin.js
- *
  * @param {Object} metadata Parsed block.json metadata.
  * @return {boolean} Is the block experimental?
  */
-module.exports = function isBlockMetadataExperimental( metadata ) {
+export default function isBlockMetadataExperimental( metadata ) {
 	return (
 		metadata &&
 		'__experimental' in metadata &&
 		metadata.__experimental !== false
 	);
-};
+}
```

Comment:
With `type: module`, this CJS module now must be converted to ESM.

In the `babel-plugin.cjs` file I'm inlining the code, because CJS modules can't import ESM modules.

We can eventually fix this by converting the Babel plugin, and most other CJS modules, to ESM. So that our codebase is fully ESM and has zero CJS files.

Reply:
> In the `babel-plugin.cjs` file I'm inlining the code, because CJS modules can't import ESM modules.

FYI, Node 20.19+, 22.12+, and 24+ allow importing ESM modules as long as they don't have any top-level `await` (directly or in any imports).

Reply:
Good to know 🙂 Gutenberg is still on v20, and with my local v20.16.0 I was getting:
```
Error [ERR_REQUIRE_ESM]: [BABEL]: require() of ES Module src/utils/is-block-metadata-experimental.js from babel-plugin.cjs not supported.
Instead change the require of is-block-metadata-experimental.js in babel-plugin.cjs to a dynamic import() which is available in all CommonJS modules.
```

Reply:
Yeah, 20.19 is the first v20 version that has it not behind a feature flag. 🙂